mitosheet/src/DataFrameMimeRenderer.tsx

```diff
@@ -0,0 +1,132 @@
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { IRenderMime } from '@jupyterlab/rendermime-interfaces';
+import { Widget } from '@lumino/widgets';
+import { ReactWidget } from '@jupyterlab/apputils';
+import * as React from 'react';
+import { INotebookTracker } from '@jupyterlab/notebook';
+import { getLastNonEmptyLine } from './jupyter/code';
+import { getCellAtIndex, getCellText } from './jupyter/extensionUtils';
+
+const CLASS_NAME = 'jp-DataFrameViewer';
+
+// React component to display the output from Python
+const SheetOutputComponent = (props: { htmlContent: string, jsCode?: string }) => {
+    // Hook to run the JavaScript code after the component is mounted
+    React.useEffect(() => {
+        if (props.jsCode) {
+            // Dynamically create a script tag to execute the JS code
+            const script = document.createElement('script');
+            script.type = 'text/javascript';
+            script.text = props.jsCode;
+            document.body.appendChild(script);
+            console.log("JavaScript code injected and executed.");
+
+            // Clean up the script after execution
+            return () => {
+                document.body.removeChild(script);
+            };
+        }
+    }, [props.jsCode]);
+
+    return (
+        <div
+            dangerouslySetInnerHTML={{ __html: props.htmlContent }}
+        ></div>
+    );
+};
+
+export class DataFrameMimeRenderer extends Widget implements IRenderMime.IRenderer {
+    private _app: JupyterFrontEnd;
+    private _notebookTracker: INotebookTracker;
+
+    constructor(app: JupyterFrontEnd, options: IRenderMime.IRendererOptions, notebookTracker: INotebookTracker) {
+        super();
+        this._app = app;
+        this.addClass(CLASS_NAME);
+        this._notebookTracker = notebookTracker;
+    }
+
+    async renderModel(model: IRenderMime.IMimeModel): Promise<void> {
+        this.node.innerHTML = '';  // Clear previous content
+        const originalRawData = model.data['text/html']?.toString();
+        const isDataframeOutput = originalRawData?.includes('class="dataframe"');
+
+
+        const notebook = this._notebookTracker.currentWidget?.content;
+        const cells = notebook?.model?.cells;
+        const activeCellIndex = notebook?.activeCellIndex
+
+        let dataframeVariableName = undefined;
+        if (activeCellIndex) {
+            const previousCell = getCellAtIndex(cells, activeCellIndex - 1)
+            dataframeVariableName = getLastNonEmptyLine(getCellText(previousCell))
```

Comment:
Test more cases: things like having two dataframes as the output, non dataframes as the output, etc. 

Add UI tests for these